article/app/services/dotcomponents/ArticlePicker.scala

```diff
@@ -0,0 +1,144 @@
+package services.dotcomponents
+
+import controllers.ArticlePage
+import model.PageWithStoryPackage
+import implicits.Requests._
+import model.liveblog.{BlockElement, TextBlockElement}
+import play.api.mvc.RequestHeader
+import views.support.Commercial
+
+object ArticlePageChecks {
+
+  def isAdFree(page: PageWithStoryPackage, request: RequestHeader): Boolean = {
+    page.item.content.shouldHideAdverts || Commercial.isAdFree(request)
+  }
+
+  def isDiscussionDisabled(page: PageWithStoryPackage): Boolean = {
+    (! page.article.content.trail.isCommentable) && page.article.content.trail.isClosedForComments
+  }
+
+  def hasBlocks(page: PageWithStoryPackage): Boolean = {
+    page.article.blocks match {
+      case Some(b) => b.body.nonEmpty
+      case None => false
+    }
+  }
+
+  def isSupportedType(page: PageWithStoryPackage): Boolean = {
+    page match {
+      case a: ArticlePage => true
+      case _ => false
+    }
+  }
+
+  def hasOnlySupportedElements(page: PageWithStoryPackage): Boolean = {
+    // See: https://github.com/guardian/dotcom-rendering/blob/master/packages/frontend/amp/components/lib/Elements.tsx
+    def unsupportedElement(blockElement: BlockElement) = blockElement match {
+      case _: TextBlockElement => false
+      case _ => true
```

Comment:
It looks like we support `ImageBlockElement` on web also

Reply:
+1